controllers/datadogagent/controller.go

```diff
@@ -131,19 +135,24 @@ func reconcilerOptionsToFeatureOptions(opts *ReconcilerOptions, logger logr.Logg
 func (r *Reconciler) reconcileInstance(ctx context.Context, logger logr.Logger, instance *datadoghqv1alpha1.DatadogAgent) (reconcile.Result, error) {
 	var result reconcile.Result
 
-	features, err := feature.BuildFeaturesV1(instance, reconcilerOptionsToFeatureOptions(&r.options, logger))
+	features, componentsEnabled, err := feature.BuildFeaturesV1(instance, reconcilerOptionsToFeatureOptions(&r.options, logger))
```

Comment:
i've been mulling over this and this pattern seems OK to me. a suggestion is to rename this to `requiredComponents` to better indicate that this is information to be used later (i presume in the reconcile<component> functions below), rather than something that's already been enabled.